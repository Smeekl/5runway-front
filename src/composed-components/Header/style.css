.icons__btn {
  border: 0;
  padding: 0;
  transition: 0.2s all;
}

.icons__btn:hover {
  opacity: 0.7;
}

.header__main {
  min-height: 70px;
  min-width: 100%;
  padding: 0 5% 0 3%;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.logo {
  margin: 0 3%;
}

.logo__icon {
  display: inline-flex;
}

.header__logo {
  transition: 0.2s all;
}

.header__logo:hover {
  opacity: 0.7;
}

.search {
  margin: 0 2.5%;
}

.icons {
  width: 4.2%;
  margin: 0 4.2%;
}

.options__items {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.options__btn {
  width: 127px;
  height: 36px;
  background: var(--color-secondary-mad-blue);
  border-radius: 15px;
  border: none;
}

.options__btn > span {
  transition: 0.2s all;
  font-weight: 500;
}

.options__btn:hover {
  background: var(--color-primary);
  border-radius: 15px;
  border: none;
}

.options__btn:hover > span {
  color: var(--color-secondary-white);
}

.options__btn > svg > path {
  transition: 0.2s all;
}

.options__btn:hover > svg > path,
.options__btn:focus > svg > path {
  fill: var(--color-secondary-white);
}

.options__btn:focus {
  background: var(--color-primary);
  border-radius: 15px;
  border: none;
}

.options__btn:focus > span {
  color: var(--color-secondary-white);
}

.options__compare {
  margin-right: 7px;
}

.icons__items {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.options {
  padding: 0.5%;
  width: 34%;
}
