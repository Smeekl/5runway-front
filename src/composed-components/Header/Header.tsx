import { Link } from 'react-router-dom';
import { ReactComponent as IconCompare } from '@svgIcons/iconCompare.svg';
import { ReactComponent as IconUser } from '@svgIcons/iconUser.svg';
import { ReactComponent as IconHeart } from '@svgIcons/iconHeart.svg';
import { ReactComponent as Logo } from '@svgIcons/iconLogo.svg';
import Button from '@components/Button/Button';
import SearchInput from '@components/SearchInput/SearchInput';
import Subheader from '@composed-components/Subheader/Subheader';
import './style.css';

const Header = () => (
  <header className="header">
    <div className="header__main">
      <Link to="/" className="header__logo logo">
        <Logo className="logo__icon" />
      </Link>
      <div className="header__search search">
        <SearchInput />
      </div>
      <div className="header__options options">
        <ul className="options__items">
          <li className="options__item">
            <Link className="options__link" to="/">
              <Button className="options__btn">Sell Now</Button>
            </Link>
          </li>
          <li className="options__item">
            <Link className="options__link" to="/">
              <Button
                className="options__btn"
                icon={<IconCompare className="options__compare" />}
              >
                Compare
              </Button>
            </Link>
          </li>
          <li className="options__item">
            <Link className="options__link" to="/">
              <Button className="options__btn">Filter</Button>
            </Link>
          </li>
        </ul>
      </div>
      <div className="header__icons icons">
        <ul className="icons__items">
          <li className="icons__item">
            <Link to="/">
              <Button className="icons__btn">
                <IconHeart />
              </Button>
            </Link>
          </li>
          <li className="icons__item">
            <Link to="/">
              <Button className="icons__btn">
                <IconUser />
              </Button>
            </Link>
          </li>
        </ul>
      </div>
    </div>
    <Subheader />
  </header>
);

export default Header;
