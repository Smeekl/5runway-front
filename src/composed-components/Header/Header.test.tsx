import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from './Header';

describe('Header UI tests', () => {
  test('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should be in the document', () => {
    const { container } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );
    expect(container).toBeInTheDocument();
  });
});
