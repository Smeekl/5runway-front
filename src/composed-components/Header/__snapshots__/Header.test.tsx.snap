// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header UI tests renders CategoriesMenuCard 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="header__main"
    >
      <a
        class="header__logo logo"
        href="/"
      >
        <reactcomponent
          class="logo__icon"
        />
      </a>
      <div
        class="header__search search"
      >
        <span
          class="ant-input-affix-wrapper search__field"
        >
          <input
            class="ant-input"
            type="text"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <reactcomponent
              class="search__icon"
            />
          </span>
        </span>
      </div>
      <div
        class="header__options options"
      >
        <ul
          class="options__items"
        >
          <li
            class="options__item"
          >
            <a
              class="options__link"
              href="/"
            >
              <button
                class="ant-btn ant-btn-default options__btn"
                type="button"
              >
                <span>
                  Sell Now
                </span>
              </button>
            </a>
          </li>
          <li
            class="options__item"
          >
            <a
              class="options__link"
              href="/"
            >
              <button
                class="ant-btn ant-btn-default options__btn"
                type="button"
              >
                <reactcomponent
                  class="options__compare"
                />
                <span>
                  Compare
                </span>
              </button>
            </a>
          </li>
          <li
            class="options__item"
          >
            <a
              class="options__link"
              href="/"
            >
              <button
                class="ant-btn ant-btn-default options__btn"
                type="button"
              >
                <span>
                  Filter
                </span>
              </button>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="header__icons icons"
      >
        <ul
          class="icons__items"
        >
          <li
            class="icons__item"
          >
            <a
              href="/"
            >
              <button
                class="ant-btn ant-btn-default icons__btn"
                type="button"
              >
                <reactcomponent />
              </button>
            </a>
          </li>
          <li
            class="icons__item"
          >
            <a
              href="/"
            >
              <button
                class="ant-btn ant-btn-default icons__btn"
                type="button"
              >
                <reactcomponent />
              </button>
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="subheader"
    >
      <ul
        class="subheader__links"
      >
        <li
          class="subheader__item"
        >
          <a
            class="subheader__link"
            href="/"
          >
            CATEGORIES
          </a>
        </li>
        <li
          class="subheader__item"
        >
          <a
            class="subheader__link"
            href="/"
          >
            TOP PRODUCT
          </a>
        </li>
        <li
          class="subheader__item"
        >
          <a
            class="subheader__link"
            href="/"
          >
            ARTICLES
          </a>
        </li>
        <li
          class="subheader__item"
        >
          <a
            class="subheader__link"
            href="/"
          >
            PRICING
          </a>
        </li>
      </ul>
    </div>
  </header>
</div>
`;
